src/scripts/app.js

```diff
@@ -1 +1,58 @@
-console.log('hello');
\ No newline at end of file
+console.log('hello');
+
+// PRODUCT SECTION RATING STAR MECHANISM -- WDP181103-19
+
+const productsSection = document.querySelector(".section--products");
+const starBoxes = document.querySelectorAll(".stars");
+
+productsSection.addEventListener("mouseover", function(e) {
+    if (event.target.classList.contains('star')) {
+        let arr = Object.values(event.relatedTarget.children);
+        let index = arr.indexOf(event.target);
+        desactiveStars();
+        for (i=0; i<=index; i++) {
+            event.relatedTarget.children[i].classList.add("active");
+        }
+    }
+    else if (event.target.classList.contains('stars')) {
+        deselectStars();
+    }   
+});
+
+productsSection.addEventListener("click", function(e) {
+    if (event.target.classList.contains('star')) {
+        let arr = Object.values(event.target.parentElement.children);
+        let index = arr.indexOf(event.target);
+        desactiveStars();
+        unmarkStars();
+        for (i=0; i<=index; i++) {
+            event.target.parentElement.children[i].classList.add("permanent");
+        }
+    }
+});
+
+// deselecting stars
+function deselectStars() {
+    if (event.target.children[0].classList.contains("permanent")){}
+    else {
+        for (i=0; i<=1; i++) {
+            event.target.children[i].classList.add("full");
+        }
+    }
+    for (i=0; i<=4; i++) {
+        event.target.children[i].classList.remove("active");
+    }
```

Comment:
Zauważyłem pewien schemat w wielu miejscach. Zobacz, że kilka razy masz taki sam kod mniej więcej:
```for (i=jakas-wartosc; i<=inna-wartosc; i++) {
        event.target.children[i].classList.remove("jakasklasa");
```

Wiadomo, ze raz zmienaisz jeden element, raz dwa itd, ale zawsze to jest pętla for i ustawianie albo usuwanie jakiejs klasy. A gdyby tak zrobic sobie dwie funkcje pomocnicze? W stylu setStarsClass(start, to, className) i analogiczną unsetStarsClass.